connector/src/main/java/org/geysermc/connector/network/translators/java/entity/JavaEntitySetPassengersTranslator.java

```diff
@@ -99,66 +101,149 @@ public void translate(ServerEntitySetPassengersPacket packet, GeyserSession sess
                 session.sendUpstreamPacket(linkPacket);
                 passengers.remove(passenger.getEntityId());
 
-                this.updateOffset(passenger, entity.getEntityType(), session, false, false, (passengers.size() > 1));
+                this.updateOffset(passenger, entity, session, false, false, (packet.getPassengerIds().length > 1));
+            } else {
+                this.updateOffset(passenger, entity, session, (packet.getPassengerIds()[0] == passengerId), true, (packet.getPassengerIds().length > 1));
             }
 
             // Force an update to the passenger metadata
             passenger.updateBedrockMetadata(session);
         }
-
-        if (entity.getEntityType() == EntityType.HORSE) {
-            entity.getMetadata().put(EntityData.RIDER_SEAT_POSITION, Vector3f.from(0.0f, 2.3200102f, -0.2f));
-            entity.getMetadata().put(EntityData.RIDER_MAX_ROTATION, 181.0f);
```

Comment:
Can you reimplement this? Without it, player rotation locking on horses is weird. 

Otherwise, I tested the PR; looks great.

Reply:
I can't seem to find any difference with or without these lines, but I've re-added them.

Reply:
Hmm, OK. I'll try to do some final testing tonight and then merge this in.